### ISSUE TITLE
ONNX转ncnn几种操作不支持

### ISSUE THREAD

**Issue by yizhaoyanbo** [Tencent/ncnn#833]
我需要将onnx模型转到ncnn。但是转换的时候提示。
Gather not supported yet!
  # axis=0
Unsqueeze not supported yet!
  # axes 7
Unsqueeze not supported yet!
  # axes 7

转换出来的文件也加载失败：
layer Shape not exists or registered
network graph not ready

有没有遇到同样问题的大大们，都是怎么处理的。感谢。

**Comment by WeiyiLi** [comment]
Maybe you need to write the supportive operation by yourself...This is what we do for tensorflow2ncnn

**Comment by nihui** [comment]
试试看 daquexian 的优化工具，可以把这些op去掉
https://github.com/daquexian/onnx-simplifier

**Comment by hhhhhhogan** [comment]
> 我需要将onnx模型转到ncnn。但是转换的时候提示。
> Gather not supported yet!
> 
> # axis=0
> Unsqueeze not supported yet!
> 
> # axes 7
> Unsqueeze not supported yet!
> 
> # axes 7
> 转换出来的文件也加载失败：
> layer Shape not exists or registered
> network graph not ready
> 
> 有没有遇到同样问题的大大们，都是怎么处理的。感谢。

请问你这里的问题解决了吗？

**Comment by LRY89757** [comment]
> 试试看 daquexian 的优化工具，可以把这些op去掉 https://github.com/daquexian/onnx-simplifier

谢谢nihuiQAQ!

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
